src/fcast.c

```diff
@@ -3,6 +3,13 @@
 // #include <signal.h> // the debugging machinery + breakpoint aidee
 // raise(SIGINT);
 
+bool any_NA_int(int N_data, int *idx){
+  for (int data_i=0; data_i<N_data; ++data_i) {
+    if(idx[data_i] == NA_INTEGER)return true;
+  }
+  return false;
+}  
+
```

Comment:
```suggestion
inline bool any_NA_int(int n, int *idx) {
  for (int i=0; i<n; ++i) {
    if (idx[i] == NA_INTEGER) return true;
  }
  return false;
}  

```
comestic and inline may better

Reply:
Yes agree `i` instead of `data_i` reads better.
The `int n` could be `const int n` too. Depending on architecture and compiler optimization abilities and flags, it can help speed to tell the compiler that the `n` is `const` so it can optimize the `i<n` computation on every loop iteration; e.g. loop unrolling. I don't really know and it's hard to test but the way I think about it is to use `const` when we can to give all compilers everywhere the best chance. The easier example is a loop condition like `i<LENGTH(x)`; that `LENGTH(x)` should be pre-saved into `const int n=LENGTH(x)` otherwise I suspect it's quite likely that `LENGTH(x)` will be recomputed on each and every iteration and that API call has overhead, albeit small. I know there's no API call in this example but even so just being in the habit of using `const` when we can might help speed and can sometimes help catch bugs by protecting that variable from writes.